src/dbnode/storage/bootstrap.go

```diff
@@ -236,25 +212,21 @@ func (m *bootstrapManager) bootstrap() error {
 		return err
 	}
 
+	instrCtx := m.instrumentation.bootstrapPreparing()
+
 	accmulators := make([]bootstrap.NamespaceDataAccumulator, 0, len(namespaces))
 	defer func() {
 		// Close all accumulators at bootstrap completion, only error
 		// it returns is if already closed, so this is a code bug if ever
 		// an error returned.
 		for _, accumulator := range accmulators {
 			if err := accumulator.Close(); err != nil {
-				instrument.EmitAndLogInvariantViolation(m.opts.InstrumentOptions(),
-					func(l *zap.Logger) {
-						l.Error("could not close bootstrap data accumulator",
-							zap.Error(err))
-					})
+				instrument.EmitAndLogInvariantViolation(instrCtx.instrumentOptions,
+					instrCtx.logFn(err, "could not close bootstrap data accumulator"))
```

Comment:
Perhaps better to just add an `emitAndLogInvariantViolation` method to the instrumentation context rather than pass it a function that takes a log (and remove the `logFn` implementation)?